plugin/secrets_service_account_key.go

```diff
@@ -232,3 +403,84 @@ would generate service account keys for the "deploy" role set.
 On the backend, each roleset is associated with a service account under
 which secrets/keys are created.
 `
+
+type serviceAccountKeyCacheCollection struct {
+	Items map[string]*serviceAccountKeyCacheItem
+}
+
+func newServiceAccountKeyCacheCollection(item *serviceAccountKeyCacheItem, itemKey string) *serviceAccountKeyCacheCollection {
+	cacheCollection := new(serviceAccountKeyCacheCollection)
+	cacheCollection.Items = make(map[string]*serviceAccountKeyCacheItem)
+
+	if item != nil && itemKey != "" {
+		cacheCollection.Items[itemKey] = item
+	}
+
+	return cacheCollection
+}
+
+func (c *serviceAccountKeyCacheCollection) getFirstValidItem(rsBindingHash string) (string, *serviceAccountKeyCacheItem) {
```

Comment:
Might want to rename the method to `getLatestItem` to make it clear what we expect the method to do

Reply:
Sure, I'll update it in the next commit.